### PR TITLE
Add macOS testing 

### LINKED ISSUE #170: Integration tests fail on MacOS
Tests run on MacBook Pro (2017) i7/16Gb/512SSD macOS Catalina 10.15.7.

Fails:

- `test_cartridge`;
- `test_connect`;
- `test_rocks`;
- `test_running`.

`test_install` freezes.

I will supplement this issue with details below.

### LINKED ISSUE #254: Add tests to macos workflow
Currently, only tt compilability is checked in the macos workflow. It is also necessary to run all the tests, in the same way as it was done for Linux. It is also necessary to support the full launch of tests on the full-ci label.

### PR DESCRIPTION
**Pull request by better0fdead**
Added macOS test workflow.
Fixed tests for macOS.

- Tests - not needed
- Changelog - not needed
- Documentation - not needed

Closes #254 
Closes #170

### CODE REVIEW

File: cli/build/build_test.go [hunk 1]
```diff
@@ -62,8 +72,18 @@ func TestFillCtxAppPathIsFile(t *testing.T) {
 	require.NoError(t, os.Chdir(workDir))
 	defer os.Chdir(wd)
 	var buildCtx BuildCtx
+
+	osType, _ := util.GetOs()
+	appDir := ""
+	switch osType {
+	case util.OsMacos:
+		appDir = filepath.Join("/private", workDir, "app1")
```
Comment (psergee) [inline cli/build/build_test.go:80]: Why do we need these paths manipulations in tests code? Why cannot we just set TMPDIR env var for our tests under Mac?

File: cli/install_ee/install_ee_test.go [hunk 2]
```diff
@@ -30,9 +31,34 @@ func TestGetVersions(t *testing.T) {
 			err:    fmt.Errorf("no packages for this OS"),
 		}
 
-	inputData1 := []byte(`<a href="/enterprise/tarantool-enterprise-bundle-` +
-		`1.10.10-52-g0df29b137-r419.tar.gz">tarantool-enterprise-bundle-1.10.10-` +
-		`52-g0df29b137-r419.tar.gz</a>                                      2021-08-18 ` +
+	arch, err := util.GetArch()
+	assert.Nil(err)
```
Comment (psergee) [inline cli/install_ee/install_ee_test.go:35]: Suppose, it must be `require`. And there is a special function for errors: NoError. It is more readable. Don't use Nil to check if no error.

File: cli/install_ee/install_ee_test.go [hunk 3]
```diff
@@ -30,9 +31,34 @@ func TestGetVersions(t *testing.T) {
...
+		if arch == "x86_64" {
+			arch = ""
+		}
+	case util.OsMacos:
+		// Bundles without specifying the architecture are all x86_64.
+		if arch == "x86_64" {
+			osName = "-macosx-"
+			arch = "x86_64"
+		} else {
+			osName = "-macosx-"
```
Comment (psergee) [inline cli/install_ee/install_ee_test.go:53]: Why duplicate 
`osName = "-macosx-"`
?
Move it out of condition.

File: cli/install_ee/install_ee_test.go [hunk 4]
```diff
@@ -30,9 +31,34 @@ func TestGetVersions(t *testing.T) {
...
+	case util.OsLinux:
+		// Bundles without specifying the architecture are all x86_64.
+		if arch == "x86_64" {
+			arch = ""
+		}
+	case util.OsMacos:
+		// Bundles without specifying the architecture are all x86_64.
+		if arch == "x86_64" {
+			osName = "-macosx-"
+			arch = "x86_64"
```
Comment (psergee) [inline cli/install_ee/install_ee_test.go:51]: Arch is already x86_64, why is this line needed?

File: cli/install_ee/install_ee_test.go [hunk 5]
```diff
@@ -30,9 +31,34 @@ func TestGetVersions(t *testing.T) {
...
+	arch, err := util.GetArch()
+	assert.Nil(err)
+
+	osType, err := util.GetOs()
+	assert.Nil(err)
+	inputData1 := []byte(``)
+	osName := ""
+	switch osType {
+	case util.OsLinux:
+		// Bundles without specifying the architecture are all x86_64.
```
Comment (psergee) [inline cli/install_ee/install_ee_test.go:43]: Is this comment related to the code below?

File: cli/build/build_test.go [hunk 6]
```diff
@@ -24,17 +25,26 @@ func TestFillCtx(t *testing.T) {
 	require.NoError(t, os.Chdir(workDir))
 	defer os.Chdir(wd)
 	var buildCtx BuildCtx
+
+	osType, _ := util.GetOs()
+	switch osType {
+	case util.OsMacos:
+		workDir = filepath.Join("/private", workDir)
```
Comment (psergee) [inline cli/build/build_test.go:32]: There is a lot of test, which uses TempDir calls. Please, add the comment why this trick is required for MacOS.

**Review by psergee — CHANGES_REQUESTED** [review]

File: cli/build/build_test.go [hunk 7]
```diff
@@ -24,17 +25,26 @@ func TestFillCtx(t *testing.T) {
 	require.NoError(t, os.Chdir(workDir))
 	defer os.Chdir(wd)
 	var buildCtx BuildCtx
+
+	osType, _ := util.GetOs()
+	switch osType {
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply cli/build/build_test.go]: Changed

File: cli/install_ee/install_ee_test.go (hunk 2 above)
Comment (better0fdead) [reply cli/install_ee/install_ee_test.go]: Changed

File: cli/install_ee/install_ee_test.go (hunk 3 above)
Comment (better0fdead) [reply cli/install_ee/install_ee_test.go]: Changed

File: cli/install_ee/install_ee_test.go (hunk 4 above)
Comment (better0fdead) [reply cli/install_ee/install_ee_test.go]: Changed

File: cli/install_ee/install_ee_test.go (hunk 5 above)
Comment (better0fdead) [reply cli/install_ee/install_ee_test.go]: Removed

File: cli/build/build_test.go (hunk 6 above)
Comment (better0fdead) [reply cli/build/build_test.go:33]: Added comment

File: cli/build/build_test.go [hunk 8]
```diff
@@ -24,17 +25,31 @@ func TestFillCtx(t *testing.T) {
...
+	if util.OsMacos == os {
+		workDir = filepath.Join("/private", workDir)
+	}
+
+	appDir := filepath.Join(workDir, "app1")
+	appDir2 := filepath.Join(workDir, "app2")
+
 	require.NoError(t, FillCtx(&buildCtx, []string{"app1"}))
-	assert.Equal(t, buildCtx.BuildDir, filepath.Join(workDir, "app1"))
+	assert.Equal(t, buildCtx.BuildDir, appDir)
```
Comment (psergee) [inline cli/build/build_test.go:43]: I don't like the idea to rely on specific symlinks target and hard-coded directory names like "/private".
Let's just check the suffix here (strings.HasSuffix workdir + app1). Or use os.Getwd result instead of tempDir for file names comparison.

File: test/integration/pack/test_pack.py [hunk 9]
```diff
@@ -631,6 +632,9 @@ def test_pack_rpm(tt_cmd, tmpdir):
 
 @pytest.mark.slow
 def test_pack_rpm_use_docker(tt_cmd, tmpdir):
+    if platform.system() == "Darwin":
+        pytest.skip("/set platform is unsupported")
```
Comment (psergee) [inline test/integration/pack/test_pack.py:636]: Why do we skip pack --use-docker for OSX? It should work AFAIK. @vr009

Comment (better0fdead) [reply test/integration/pack/test_pack.py]: Removed

File: cli/install_ee/install_ee_test.go (hunk 2 above)
Comment (psergee) [reply cli/install_ee/install_ee_test.go]: Please use require instead of assert here. Suppose, the test should not continue if GetArch or GetOs fails.

File: cli/build/build_test.go (hunk 8 above)
Comment (better0fdead) [reply cli/build/build_test.go:34]: Changed to os.Getwd file names comparison

**Review by psergee — APPROVED** [review]

**Review by 0x501D — APPROVED** [review]
